src/modelinstance.cpp

```diff
@@ -505,11 +505,8 @@ Status ModelInstance::reloadModel(const ModelConfig& config, const DynamicModelP
 
 Status ModelInstance::recoverFromReloadingError(const Status& status) {
     if (status == StatusCode::RESHAPE_ERROR) {
-        auto recoveryStatus = this->recoverFromReshapeError();
-        if (!recoveryStatus.ok()) {
-            return recoveryStatus;
-        }
-        return status;
+        SPDLOG_DEBUG("Recovering model {} version {}. Unloading", getName(), getVersion());
+        unloadModel();
```

Comment:
This is most likely incomplete -> it does not recover now - just unloads the model.